CMakeLists.txt

```diff
@@ -1,91 +1,59 @@
-    find_package(PythonLibsNew 3.6 REQUIRED)
-    message(STATUS "${Cyan}Found Python ${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}${ColourReset}: ${PYTHON_EXECUTABLE} (found version ${PYTHON_VERSION_STRING})")
-    set(pybind_ext "pybind11_external")
-    set(python_flags
-              -DPYTHON_EXECUTABLE=${PYTHON_EXECUTABLE}
-              -DPYTHON_INCLUDE_DIR=${PYTHON_INCLUDE_DIR}
-              -DPYTHON_LIBRARY=${PYTHON_LIBRARY}
-        )
+# if CMAKE_BUILD_TYPE undefined, we set it to Release
+if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-include(GNUInstallDirs)
-include(autocmake_safeguards)
-include(custom_color_messages)
-include(ExternalProject)
-
-add_subdirectory(external/upstream)
-
-ExternalProject_Add(cppe-core
-   DEPENDS ${pybind_ext}
-   SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cppe
-   CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
-              -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-              -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-              -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-              -DCMAKE_CXX_STANDARD=${cppe_CXX_STANDARD}
-              -DCMAKE_CXX_STANDARD_REQUIRED=ON
-              -DEIGEN3_INCLUDE_DIRS=${PROJECT_SOURCE_DIR}/external/eigen3
-              -DENABLE_OPENMP=${ENABLE_OPENMP}
-              -DENABLE_PYTHON_INTERFACE=${ENABLE_PYTHON_INTERFACE}
-              ${python_flags}
-              -Dpybind11_DIR=${pybind11_DIR}
-              -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
-              -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
-              -DCMAKE_POSITION_INDEPENDENT_CODE=${BUILD_FPIC}
-              -DINSTALL_DEVEL_HEADERS=${INSTALL_DEVEL_HEADERS}
-              -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
-              -DPYMOD_INSTALL_LIBDIR=${PYMOD_INSTALL_LIBDIR}
-              -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
-   CMAKE_CACHE_ARGS -DCMAKE_PREFIX_PATH:PATH=${CMAKE_PREFIX_PATH}
-   BUILD_ALWAYS 1)
+# options handling utilities
+include(CMakeDependentOption)
+# Macro for printing an option in a consistent manner
+# Written by Lori A. Burns (@loriab) and Ryan M. Richard (@ryanmrichard)
+# Syntax: print_option(<option to print> <was specified>)
+function(print_option variable default)
+  if(NOT DEFINED ${variable} OR "${${variable}}" STREQUAL "")
+    message(STATUS "Setting (unspecified) option ${variable}: ${default}")
+  else()
+    message(STATUS "Setting option ${variable}: ${${variable}}")
+  endif()
+endfunction()
+
+# Wraps an option with default ON/OFF. Adds nice messaging to option()
+# Written by Lori A. Burns (@loriab) and Ryan M. Richard (@ryanmrichard)
+# Syntax: option_with_print(<option name> <description> <default value>)
+macro(option_with_print variable msge default)
+  print_option(${variable} ${default})
+  option(${variable} ${msge} ${default})
+endmacro()
+
+# Wraps an option with a default other than ON/OFF and prints it
+# Written by Lori A. Burns (@loriab) and Ryan M. Richard (@ryanmrichard)
```

Comment:
Ryan resigned credit long ago (I asked him), and I don't need it either. So feel free to declutter if you want.